### ISSUE TITLE
Focused Launch: Ensure selected domain is always shown to the user in the Summary View

### ISSUE THREAD

**Issue by tjcafferkey** [Automattic/wp-calypso#47540]
**Important**: Potentially blocked by https://github.com/Automattic/wp-calypso/issues/47512 as both detail with changes to the `<DomainPicker />` which will be used in both solutions.

### What

There are a few scenarios under which the user can visit the Summary view while having selected a domain that is not part of the 3 suggested items in the Domain Step:

- Scenario A. The user enters the site title, picks a domain from the Domain Picker in the Domain step, and then changes the site title again. The Domain step shows new recommendations, and the currently selected domain is not shown to the user anymore.
- Scenario B. The user visits the Domain Details view, picks a domain that is not part of the initial suggestions in the Summary view. The user navigates back to the Summary view, and the currently selected domain is not shown to the user.

### Why

We need to always show the currently selected domain to the user in the Summary view, while keeping the top paid suggestions always at the top and free suggestion always at the bottom.

### Acceptance Criteria

- [ ] ~The first 2 items shown in the Domain Picker in the Summary view should always be the top 2 suggested paid domains (the first item should have a "recommended" label).~ Moved this to https://github.com/Automattic/wp-calypso/issues/47748
- [ ] ~The last item shown in the Domain Picker in the Summary view should always be the default, free WordPress subdomain suggestion~ Moved this to https://github.com/Automattic/wp-calypso/issues/47748
- [x] If the currently selected domain matches one of the 3 items defined in the points above, the Domain Picker in the Summary view should highlight the matching item
- [x] If the currently selected domain doesn't match any of the 3 items defined in the points above, the Domain Picker in the Summary view should display an additional (highlighted) item. This domain item should be the 3rd item from the top — after the top 2 paid suggestions, but before the free domain suggestion.
- [x] If the additional domain item (spec'd in the point above) is showing in the Domain Picker in the Summary view, and the user selects another domain, the additional domain item should not disappear from the Domain Picker until the Summary view is re-visited.

<img width="1042" alt="Screen Shot 2020-11-13 at 9 40 53 am" src="https://user-images.githubusercontent.com/67085494/99009798-5ff1c600-2594-11eb-8f98-082c75bf42d1.png">

### Considerations

For added context, if the user hasn't made an explicit selection yet in the context of the launch flow, the domain step should not pre-select any domain (as specc'd and discussed separately in #47458)

* This issue may require changes to the domain picker package, we need to ensure this doesn't negatively impact the areas it's used in the following:
	- Focused Launch
	- Step-by-step Launch
	- New onboarding

**Comment by tjcafferkey** [comment]
@ollierozdarz just seen your comment [here](https://github.com/Automattic/wp-calypso/issues/47458#issuecomment-729430634) about no domains (or plans) being pre-selected for the user. I am assuming that is the case for if this is the first time the user views the focused launch modal.

There are some other considerations outlined in this issue that me and @ciampo, would you mind reading the issue above and let me know if this sounds right to you?

**Comment by ollierozdarz** [comment]
Hey @tjcafferkey, the acceptance criteria for this one looks good to me!
